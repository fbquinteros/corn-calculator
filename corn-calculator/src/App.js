import logo from './logo.svg';
import './App.css';

function App() {
  return (
    <div className="App">
      <header className="App-header">        
        <h1>
          Corn calculator
        </h1>
      <label forhtml="quantity">Quantity:</label>
      <input type="number" name="quantity"></input>      
      </header>
    </div>
  );
}

export default App;
