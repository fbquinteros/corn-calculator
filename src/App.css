header {
  padding: 80px;
  text-align: center;
  background: #1abc9c;
  color: white;
}

.header h1 {
  font-size: 40px;
}

html {
  font-family: Verdana, sans-serif;
  font-size: 15px;
  line-height: 1.5;
  font-weight: normal;
}

h1,
h2,
h3,
h4,
h5,
h6 {
  font-family: "Segoe UI", Verdana, sans-serif;
  font-weight: 400;
  line-height: 1;
  margin: 20px 0;
}

h1 a,
h2 a,
h3 a,
h4 a,
h5 a,
h6 a {
  font-weight: inherit;
}
h1 {
  font-size: 36px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 20px;
}
h5 {
  font-size: 18px;
}
h6 {
  font-size: 16px;
}

hr {
  height: 0;
  border: 0;
  border-top: 1px solid #eee;
  margin: 20px 0;
}

legend {
  background-color: gray;
  color: white;
}

fieldset {
  display: block;
  margin-left: 2px;
  margin-right: 2px;
  padding-top: 0.35em;
  padding-bottom: 0.625em;
  padding-left: 0.75em;
  padding-right: 0.75em;
  border: 2px groove (internal value);
}

.input-group  {
  margin-top:10px;
  width:60%;
  display:flex;
  justify-content:space-between;
  flex-wrap:wrap;
}

.user-info {
  font-weight: bold;
  font-style: italic;
}

form {
  margin:0 auto;
  padding:20px;
}

label, input {
  flex-basis:200px;
}

.App {
  margin: auto;
  text-align: center;
}

.App-header {
  background-color: #282c34;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  color: white;
}

.App-content {
  padding-top: 12px !important;
  padding-bottom: 12px !important;
}
