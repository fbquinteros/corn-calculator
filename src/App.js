import React, { useState } from "react";

import "./App.css";

function App() {
  const [cornQuantity, setCornQuantity] = useState(1);
  const costPerBag = 0.25;

  const cornQuantityHandler = (e) => {
    const value = Number.parseInt(e.target.value, 10) || 0;
    console.log("Quantity of corn", value);
    setCornQuantity(value);
  };

  return (
    <div className="App">
      {/* TODO: Extract Header Component */}
      <header className="App-header">
        <h1>River trip adviser</h1>
      </header>
      <div className="App-content">
        {/* TODO: Extract Trip Calculator Component */}        
        <label forhtml="cornQuantity">Corn Quantity:</label>
        <input
          type="number"
          id="cornQuantity"
          name="quantity"
          value={cornQuantity}
          onChange={cornQuantityHandler}
        />
        <br/>
        <label>Â£ {parseFloat(cornQuantity * costPerBag).toFixed(2)}</label>
      </div>
    </div>
  );
}

export default App;
