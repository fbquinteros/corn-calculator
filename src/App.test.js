import { render, screen } from '@testing-library/react';

import App from './App';

describe('App', () => {
  test('should contain a header with corn calculator', () => {
    render(<App />);
    const linkElement = screen.getByText(/Corn calculator/i);
    expect(linkElement).toBeInTheDocument();
  });

  test('should contain a quantity label', () => {
    render(<App />);
    const linkElement = screen.getByText(/Quantity/i);
    expect(linkElement).toBeInTheDocument();
  });

  test('should contain a quantity with initial default value of 1', () => {
    const component = render(<App />);
    
    const inputElement = component.container.querySelector('input');

    expect(inputElement).toBeTruthy();
    expect(inputElement.value).toBe("1");
  });

  test('should calculate the cost of 1 bag of corn at 25 pence', () => {
    render(<App />);
    
    const linkElement = screen.getByText(/Â£ 0.25/i);
    
    expect(linkElement).toBeInTheDocument();
  });

  test('should get default app in the expected html format', () => {
    const component = render(<App />);
          
    expect(component.container).toMatchSnapshot();
  });

});
