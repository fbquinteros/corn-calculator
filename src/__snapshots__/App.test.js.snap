// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should get default app in the expected html format 1`] = `
<div>
  <div
    class="App"
  >
    <header
      class="App-header"
      style="background-image: url(/farm.jpg); background-position: center; background-size: cover; background-repeat: no-repeat;"
    >
      <h1>
        River trip adviser
      </h1>
    </header>
    <div
      class="App-content"
    >
      <label
        forhtml="cornQuantity"
      >
        Corn Quantity:
      </label>
      <input
        id="cornQuantity"
        name="quantity"
        type="number"
        value="1"
      />
      <br />
      <label>
        Â£ 
        0.25
      </label>
    </div>
  </div>
</div>
`;
